v4/inprocess-proxy/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import unleash, { ApiUser, LogLevel } from 'unleash-server';
+import { createProxy } from './create-proxy';
+import winston from 'winston';
+import crypto from 'crypto';
+
+const logger = winston.createLogger({
+    level: 'info',
+    format: winston.format.json(),
+    transports: [new winston.transports.Console()],
+});
+export function start(): void {
+    unleash.start({
+        // @ts-ignore
+        getLogger: (name) => logger.child({ name }),
+        db: {
+            user: 'unleash_user',
+            password: 'passord',
+            host: 'localhost',
+            port: 5432,
+            database: 'unleashwithproxy',
+            ssl: false,
+        },
+        server: {
+            enableRequestLogger: true,
+            baseUriPath: '',
+        },
+        preHook: (app) => {
+            let devKey = crypto.randomBytes(32).toString('hex');
+            createProxy({
+                environment: 'development',
+                unleashServer: app,
+                logger,
+                clientKey: devKey,
+                ApiUser: ApiUser,
+            });
+            logger.warn(
+                `Development proxy is now available at http://localhost:4242/api/development/proxy with key: ${devKey}`,
+            );
+
+            let prodKey = crypto.randomBytes(32).toString('hex');
+            createProxy({
+                environment: 'production',
+                unleashServer: app,
+                logger,
+                ApiUser: ApiUser,
+                clientKey: prodKey,
+            });
```

Comment:
This is what creates the proxies, right? I'd love to see that called out and explained in the readme. Something like:

> ## Explanation / how it's done
> 
> To run both proxies in-process, we create a `createProxy` function that [...]
> [...]
> Then, when we start the app, we call it once for each environment configuration that we want (see the `start` function in the `src/index.ts` file.

Plus anything else you think is relevant 🤓